python/ray/dashboard/modules/train/train_head.py

```diff
@@ -92,6 +94,44 @@ async def get_train_runs(self, req: Request) -> Response:
             content_type="application/json",
         )
 
+    async def _add_actor_status(self, train_runs):
+        try:
+            from ray.train._internal.state.schema import ActorStatusEnum, RunStatusEnum
+        except ImportError:
+            logger.exception(
+                "Train is not installed. Please run `pip install ray[train]` "
+                "when setting up Ray on your cluster."
+            )
+
+        actor_status_table = {}
+        try:
+            logger.info("Getting all actor info from GCS.")
+            request = gcs_service_pb2.GetAllActorInfoRequest()
+            reply = await self._gcs_actor_info_stub.GetAllActorInfo(request, timeout=5)
+            if reply.status.code == 0:
+                for message in reply.actor_table_data:
+                    actor_table_data = actor_table_data_to_dict(message)
+                    actor_status_table[actor_table_data["actorId"]] = actor_table_data[
+                        "state"
+                    ]
+        except Exception:
+            logger.exception("Error Getting all actor info from GCS.")
+
+        for train_run in train_runs.values():
+            for worker_info in train_run.workers:
+                worker_info.status = actor_status_table.get(worker_info.actor_id, None)
+
+            # If the controller died but the run status is not updated,
+            # mark the train run as aborted
+            controller_actor_status = actor_status_table.get(
+                train_run.controller_actor_id, None
+            )
+            if (
+                controller_actor_status == ActorStatusEnum.DEAD
+                and train_run.run_status == RunStatusEnum.STARTED
+            ):
+                train_run.run_status = RunStatusEnum.ABORTED
```

Comment:
nit: Feels a bit messy to update the run status here. If we keep this here I'd at least rename the method to `_add_actor_and_update_run_status` or add a docstring that documents this behavior.

Reply:
This one is doing some post processing to handle some abnormally terminated cases. Let me update the function name here.